recipes/flatcc/all/test_package/CMakeLists.txt

```diff
@@ -1,32 +1,33 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.1)
 project(flatcc_example)
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup()
 
-set(INC_DIR "${PROJECT_SOURCE_DIR}/include")
-set(GEN_DIR "${CMAKE_CURRENT_BINARY_DIR}/generated")
-set(FBS_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
 
-include_directories("${GEN_DIR}" "${INC_DIR}")
+set(FBS_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
 
-add_executable(monster monster.c)
 
 #On MacOS System Integrity Protection (SIP) will clear the DYLD_LIBRARY_PATH variable.
 #As a result calling flatcc from cmake will currently not work if the flatcc executable
 # is linked shared. As a workaround we generate the flatbuffer C files in the Conan recipe
 # when on MacOS and flatcc option 'shared' is True.
-if (NOT MACOS_SIP_WORKAROUND)
-    add_custom_target(gen_monster_fbs ALL)
-    add_custom_command (
-        TARGET gen_monster_fbs
-        COMMAND cmake -E make_directory "${GEN_DIR}"
-        COMMAND flatcc -a -o "${GEN_DIR}" "${FBS_DIR}/monster.fbs"
-        DEPENDS flatcc "${FBS_DIR}/monster.fbs"
+if (MACOS_SIP_WORKAROUND)
+    set(INC_DIR "${PROJECT_SOURCE_DIR}/include")
+    set(GEN_DIR "${CMAKE_CURRENT_BINARY_DIR}/generated")
+    include_directories("${GEN_DIR}" "${INC_DIR}")
```

Comment:
Is this SIP workaround still needed for my version?

Reply:
Flatcc depends on the flatcc compiler shared lib when configuring with BUILD_SHARED_LIBS=On. The SIP workaround is only used then:
`if tools.os_info.is_macos and self.options["flatcc"].shared:`

Its very well possible that the workaround is not needed anymore now that RPATH is set but at the moment I don't have access to a Mac that is recent enough to have SIP. I can check next week at work.

Reply:
Please do, or just let c3i test it for you :stuck_out_tongue: 
SIP causes problems only when it needs external `DYLD_LIBRARY_PATH`, which is not the case here.

Unless cmake is configured to not set RPATH on install, which is possible:
https://cmake.org/cmake/help/latest/variable/CMAKE_SKIP_INSTALL_RPATH.html